android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java

```diff
@@ -144,40 +153,87 @@ public Bundle getSkuDetails(final int apiVersion, final String packageName, fina
+    Bundle bundleResponse = buildEmptyBundle();
     if (WalletUtils.hasWalletInstalled()) {
       try {
-        return serviceAppcoinsBilling.getPurchases(apiVersion, packageName, type, null);
+        return serviceAppcoinsBilling.getPurchases(apiVersion, packageName, type,
+            continuationToken);
       } catch (RemoteException e) {
         e.printStackTrace();
         bundleResponse.putInt(Utils.RESPONSE_CODE, ResponseCode.SERVICE_UNAVAILABLE.getValue());
       }
     } else {
+      String walletId = getWalletId();
+      if (walletId != null && type.equalsIgnoreCase("INAPP")) {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        BillingRepository billingRepository =
+            new BillingRepository(new BdsService(BuildConfig.HOST_WS, 30000));
+        GuestPurchasesInteract guestPurchaseInteract =
+            new GuestPurchasesInteract(billingRepository);
 
-      bundleResponse.putInt(Utils.RESPONSE_CODE, ResponseCode.OK.getValue());
-      bundleResponse.putStringArrayList("INAPP_PURCHASE_ITEM_LIST", new ArrayList<String>());
-      bundleResponse.putStringArrayList("INAPP_PURCHASE_DATA_LIST", new ArrayList<String>());
-      bundleResponse.putStringArrayList("INAPP_DATA_SIGNATURE_LIST", new ArrayList<String>());
+        guestPurchaseInteract.mapGuestPurchases(bundleResponse, walletId, packageName, type,
+            countDownLatch);
+        waitForPurchases(countDownLatch);
+      }
     }
-
     return bundleResponse;
   }
 
   @Override public int consumePurchase(int apiVersion, String packageName, String purchaseToken) {
-    if (WalletUtils.hasWalletInstalled()) {
-      try {
+    try {
+      if (WalletUtils.hasWalletInstalled()) {
         return serviceAppcoinsBilling.consumePurchase(apiVersion, packageName, purchaseToken);
-      } catch (RemoteException e) {
-        e.printStackTrace();
-        return ResponseCode.SERVICE_UNAVAILABLE.getValue();
+      } else {
+        String walletId = getWalletId();
+        if (walletId != null && apiVersion == SUPPORTED_API_VERSION) {
+          return consumeGuestPurchase(walletId, packageName, purchaseToken);
+        } else {
+          return ResponseCode.OK.getValue();
+        }
       }
-    } else {
-      return ResponseCode.OK.getValue();
+    } catch (RemoteException e) {
+      e.printStackTrace();
+      return ResponseCode.SERVICE_UNAVAILABLE.getValue();
     }
   }
 
+  private int consumeGuestPurchase(String walletId, String packageName, String purchaseToken) {
+    int[] responseCode = new int[1]; //Generic error
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+    BillingRepository billingRepository =
+        new BillingRepository(new BdsService(BuildConfig.HOST_WS, 30000));
+    GuestPurchasesInteract guestPurchaseInteract = new GuestPurchasesInteract(billingRepository);
+
+    guestPurchaseInteract.consumeGuestPurchase(walletId, packageName, purchaseToken, responseCode,
+        countDownLatch);
```

Comment:
i think that you shouldn't send this countDownLatch here, it's not Interactor's responsability to deal this the syncronization, if it is, it should handle it by itself